assemblyscript/assemblyscript.go

```diff
@@ -178,34 +180,41 @@ func (a *assemblyscript) abort(
 	panic(sys.NewExitError(mod.Name(), exitCode))
 }
 
-// trace implements the same named function in AssemblyScript. Ex.
+// traceNoop implements trace in wasm to avoid host call overhead on no-op.
+var traceNoop = &wasm.Func{
```

Comment:
this is declaring a wasm definition including its inlined type. this is only used by HostModule and is how functions can end up compiled to wasm even in a host module.